### DISCUSSION TITLE
VCFs with Invariant/Monomorphic Sites

### DISCUSSION THREAD

**Discussion by ksamuk** [tskit-dev/tskit#2419] [General]
Apologies if this was answered elsewhere, but is it currently possible for write_vcf() to output a VCF that includes invariant/monomorphic sites? Any suggestions would be greatly appreciated and thanks for creating/maintaining this amazing software!

**Comment by jeromekelleher** [comment]
Hi @ksamuk :wave:

The easiest way to add a monomorphic site is to just add a site that doesn't have any mutations, e.g.:
```python
ts = msprime.sim_ancestry(3, random_seed=1, sequence_length=100)

tables = ts.dump_tables()
tables.sites.add_row(position=20, ancestral_state="A")
ts = tables.tree_sequence()
print(ts.as_vcf())
```
gives
```
##fileformat=VCFv4.2
##source=tskit 0.5.2.dev0
##FILTER=<ID=PASS,Description="All filters passed">
##contig=<ID=1,length=100>
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  tsk_0   tsk_1   tsk_2
1       20      0       A       .       .       PASS    .       GT      0|0     0|0     0|0
```

(This just has a monomorphic site for simplicity, but you'd probably want to add one to a simulation with some other mutations.)

**Comment by ksamuk** [comment]
Fantastic, thank you @jeromekelleher!
